idaes/models/properties/modular_properties/eos/tests/test_ceos_PR.py

```diff
@@ -1065,3 +1074,310 @@ def test_vol_mol_phase_comp(m):
             for j in m.params.component_list
         )
     )
+
+
+class TestCEOSCriticalProps:
+    @pytest.fixture(scope="class")
+    def model(self):
+        m = ConcreteModel()
+
+        # Dummy params block
+        m.params = GenericParameterBlock(
+            components={
+                "a": {
+                    "parameter_data": {
+                        "omega": 0.1,
+                        "compress_fact_crit": 1,
+                        "dens_mol_crit": 1,
+                        "pressure_crit": 1e5,
+                        "temperature_crit": 100,
+                    }
+                },
+                "b": {
+                    "parameter_data": {
+                        "omega": 0.2,
+                        "compress_fact_crit": 1,
+                        "dens_mol_crit": 1,
+                        "pressure_crit": 2e5,
+                        "temperature_crit": 200,
+                    }
+                },
+                "c": {
+                    "parameter_data": {
+                        "omega": 0.3,
+                        "compress_fact_crit": 1,
+                        "dens_mol_crit": 1,
+                        "pressure_crit": 3e5,
+                        "temperature_crit": 300,
+                    }
+                },
+            },
```

Comment:
Here we might start to run into problems with the critical properties not being consistent. Knowing how the cubic EoS works, I imagine what happens is that T_crit and P_crit influence the mixture critical properties while rho_mol_crit and Z_crit do not.